### PR TITLE
Remove Simd.Js Code

### LINKED ISSUE #3274: Remove Simd.js code from ChakraCore - Part 2 (Remove)
Part 1 (#3273) of this change is to ifdef the code to not build. Once we figure out what can be reused for WASM simd we should remove rest of the code

### PR DESCRIPTION
**Pull request by Cellule**
Closes #3274

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/4541)

### CODE REVIEW

**Comment by Cellule** [comment]
FYI @arunetm

**Comment by Cellule** [comment]
Mainly removed `#if ENABLE_SIMDJS` code, but also code related to typespec and asm.js frontend

File: lib/Runtime/Language/WAsmjsUtils.cpp [hunk 1]
```diff
@@ -289,6 +289,9 @@ template<> Types RegisterSpace::GetRegisterSpaceType<AsmJsSIMDValue>(){return WA
             }
         }
 
+        // The offset currently carries the total size of the funcInfo after handling the last type
+        funcInfo->SetTotalSizeinBytes(offset);
+
```
Comment (agarwal-sandeep) [inline lib/Runtime/Language/WAsmjsUtils.cpp]: nit: This is some bug fix right? nothing to do with SIMD

**Comment by agarwal-sandeep** [comment]
>         // double return

do we need a break here? in case Wasm::Simd::IsEnabled() is false is it ok to fall through to double?

---
Refers to: lib/Runtime/Language/InterpreterStackFrame.cpp:2108 in 5469106. [](commit_id = 54691067a4d2ff1fb6198124f0363e88d183a040, deletion_comment = False)

File: lib/Runtime/Language/WAsmjsUtils.cpp (hunk 1 above)
Comment (Cellule) [reply lib/Runtime/Language/WAsmjsUtils.cpp]: We used to assume simd to be the last type in asm.js/wasm. With this we recalculated the size in bytes.
Without simd, it makes more sense to cache the size at the time we calculate everything and just give it back afterward, without having to know which type is last nor duplicating the math.
See [AsmJsFunctionInfo::GetTotalSizeinBytes()'s implementation change for details](https://github.com/Microsoft/ChakraCore/pull/4541/commits/7806587638d62610232e23d66e263e212fbf697b#diff-3c5dbfb193356747438af5fa1e350b69L921)

**Comment by Cellule** [comment]
> do we need a break here? in case Wasm::Simd::IsEnabled() is false is it ok to fall through to double?

You're right. It doesn't really make sense to have a runtime check here. We'll get bad result is wasm.simd is disabled.
I checked that we don't leak any kind of information in case we get it wrong so I just added an `Assert` instead. The code is cleaner now as well

**Comment by dilijev** [comment]
Any reason someone shouldn't adopt this and get it merged?

**Comment by Cellule** [comment]
Just waiting on sign off

**Comment by Krovatkin** [comment]
@Cellule I'm very sorry about that! I completely missed it. I'll try to review it ASAP. I'm also tagging @arunetm He could also assign a back-up reviewer. FYI @Cellule I moved to a different team, so it might be a good idea to also assign @arunetm as a reviewer for all SIMD-related items in case I miss github notifications.

**Comment by Cellule** [comment]
ping @agarwal-sandeep

**Comment by arunetm** [comment]
LGTM

**Pull request merged**